packages/core/src/metadata.ts

```diff
@@ -186,15 +190,20 @@ class Metadata implements IMetadata {
     tssNonce?: number;
     tssPolyCommits?: Point[];
     factorPubs?: Point[];
+    accountIndex?: number;
     factorEncs?: {
       [factorPubID: string]: FactorEnc;
     };
+    chainCode?: string;
   }): void {
-    const { tssTag, tssNonce, tssPolyCommits, factorPubs, factorEncs } = tssData;
+    const { tssTag, tssNonce, tssPolyCommits, factorPubs, factorEncs, chainCode } = tssData;
     if (tssNonce !== undefined) this.tssNonces[tssTag] = tssNonce;
     if (tssPolyCommits) this.tssPolyCommits[tssTag] = tssPolyCommits;
     if (factorPubs) this.factorPubs[tssTag] = factorPubs;
     if (factorEncs) this.factorEncs[tssTag] = factorEncs;
+    if (chainCode && !this.chainCode) {
```

Comment:
@himanshuchawla009 , Should we throw when chainCode exist and we try to update the chainCode?
or just ignore the update